projects/core/src/lib/operators/dialogs/column-range-filter/column-range-filter.component.ts

```diff
@@ -1,17 +1,38 @@
 import {Component, OnInit, ChangeDetectionStrategy, OnDestroy} from '@angular/core';
 import {ResultTypes} from '../../result-type.model';
-import {UntypedFormArray, UntypedFormBuilder, UntypedFormGroup, Validators} from '@angular/forms';
-import {Observable, of, ReplaySubject, Subscription} from 'rxjs';
+import {FormArray, FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {combineLatest, Observable, of, ReplaySubject, Subscription} from 'rxjs';
 import {ProjectService} from '../../../project/project.service';
 import {map, mergeMap} from 'rxjs/operators';
 import {Layer, VectorLayer} from '../../../layers/layer.model';
 import {VectorLayerMetadata} from '../../../layers/layer-metadata.model';
 import {VectorColumnDataType, VectorColumnDataTypes} from '../../datatype.model';
-import {WorkflowDict} from '../../../backend/backend.model';
+import {UUID, WorkflowDict} from '../../../backend/backend.model';
 import {ClusteredPointSymbology, PointSymbology} from '../../../layers/symbology/symbology.model';
 import {colorToDict} from '../../../colors/color';
 import {RandomColorService} from '../../../util/services/random-color.service';
-import {ColumnRangeFilterDict} from '../../../backend/operator.model';
+import {ColumnRangeFilterDict, HistogramDict, HistogramParams} from '../../../backend/operator.model';
+import {VegaChartData} from '../../../plots/vega-viewer/vega-viewer.component';
+import {MapService} from '../../../map/map.service';
+import {BackendService} from '../../../backend/backend.service';
+import {UserService} from '../../../users/user.service';
+import {extentToBboxDict} from '../../../util/conversions';
+
+interface ColumnRangeFilterForm {
+    layer: FormControl<Layer | null>;
+    name: FormControl<string>;
+    filters: FormArray<FormGroup<FilterForm>>;
+}
+
+interface FilterForm {
+    attribute: FormControl<string>;
+    ranges: FormArray<FormGroup<RangeForm>>;
+}
+
+interface RangeForm {
+    min: FormControl<string>;
+    max: FormControl<string>;
+}
```

Comment:
When ranges are not specified, the create button is shown.
Users get an error when they try to click it.

Since you've specified validation conditions within the form, please disable the button when you cannot click it.

Reply:
df2d1d2
I did not specify the validation conditions. But I have now changed the validation. now validators are specified directly with the FormControls.